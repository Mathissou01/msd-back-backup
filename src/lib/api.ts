import qs from "qs";

export function getStrapiURL(path = "") {
  return `${process.env.NEXT_PUBLIC_API_URL}${path}`;
}

export function getAzureSearchURL(path = "") {
  return `${process.env.NEXT_PUBLIC_AZURE_SEARCH_URL}${path}`;
}

export async function fetchAPI(
  path: string,
  urlParamsObject = {},
  options = {},
) {
  const mergedOptions = {
    headers: {
      "Content-Type": "application/json",
    },
    ...options,
  };

  const queryString = qs.stringify(urlParamsObject);
  const requestUrl = `${getStrapiURL(
    `/api${path}${queryString ? `?${queryString}` : ""}`,
  )}`;

  const response = await fetch(requestUrl, mergedOptions);

  if (!response.ok) {
    console.error(response.statusText);
    throw new Error(`An error occurred please try again`);
  }
  return await response.json();
}

interface IFetchGraphQLParams {
  query: {};
  baseUrl?: string;
  variables?: {};
  urlParamsObject?: {};
  options?: {};
}

export async function fetchGraphQL(params: IFetchGraphQLParams) {
  const mergedOptions = {
    method: "POST",
    headers: {
      Authorization: `bearer ${process.env.STRAPI_API_TOKEN}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: params.query,
      variables: params.variables,
    }),
    ...params.options,
  };

  const queryString = qs.stringify(params.urlParamsObject);
  const requestUrl = getStrapiURL(
    `/graphql${queryString ? `?${queryString}` : ""}`,
  );

  return await fetch(requestUrl, mergedOptions).then((response) => {
    if (!response.ok) {
      throw new Error(`An error occurred please try again`);
    }
    return response.json();
  });
}

export async function fetchGraphQLDynamic(params: IFetchGraphQLParams) {
  const mergedOptions = {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      query: params.query,
      variables: params.variables,
    }),
    ...params.options,
  };

  const queryString = qs.stringify(params.urlParamsObject);
  const requestUrl = `${
    params.baseUrl ?? process.env.NEXT_PUBLIC_BASE_URL ?? ""
  }/api/graphql${queryString ? `?${queryString}` : ""}`;

  console.log(queryString);
  console.log(mergedOptions);
  const response = await fetch(requestUrl, mergedOptions);

  if (!response.ok) {
    console.error(response.statusText);
    throw new Error(`An error occurred please try again`);
  }
  return await response.json();
}

export async function fetchAzureSearch(
  path: string,
  urlParamsObject = {},
  options = {},
) {
  // Merge default and user options
  const mergedOptions = {
    headers: {
      "Content-Type": "application/json",
      "api-key": "C91tu6oXTbJ9mdQMANN96N6skIsSPKTlbakF3sNpTuAzSeDe9I1T",
    },
    ...options,
  };

  const queryString = qs.stringify(urlParamsObject, { encode: false });
  const requestUrl = `${getAzureSearchURL(
    `${path}${queryString ? `?${queryString}` : ""}`,
  )}`;

  const response = await fetch(requestUrl, mergedOptions);

  if (!response.ok) {
    console.error(response.statusText);
    throw new Error(`An error occurred please try again`);
  }
  return await response.json();
}
