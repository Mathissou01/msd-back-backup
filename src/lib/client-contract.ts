import { InputMaybe, ServiceType } from "../graphql/codegen/generated-types";

export interface IClientFields {
  clientName: string;
  clientType: string;
  siretNumber?: number;
  contactFirstName: string;
  contactLastName: string;
  contactEmail: string;
  contactPhoneNumber: string;
  isRvFrance: boolean;
  isNonExclusive: boolean;
  ccap?: number;
  clear?: number;
  isFreemium: boolean;
  servicesToActivate: Array<InputMaybe<ServiceType>> | InputMaybe<ServiceType>;
}
