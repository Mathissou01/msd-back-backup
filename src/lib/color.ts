import chroma from "chroma-js";

export interface IDefaultColorPalette {
  contrastDefaultLight: string;
  contrastDefaultDark: string;
  contrastText: string;
  primaryColor: string;
  primaryColorDark: string;
  primaryColorLight: string;
  secondaryColor: string;
  secondaryColorDark: string;
  secondaryColorLight: string;
}

export interface IColorPalette {
  contrastText?: string;
  primaryColor?: string;
  primaryColorDark?: string;
  primaryColorLight?: string;
  secondaryColor?: string;
  secondaryColorDark?: string;
  secondaryColorLight?: string;
}

const contrastDefaultLight = "#FAF8F4";
const contrastDefaultDark = "#030f40";
const contrastText = contrastDefaultDark;
const primaryColor = "#9BCD41";
const primaryColorDark = makeDarkerColor(primaryColor);
const primaryColorLight = makeLighterColor(primaryColor);
const secondaryColor = "#FFC229";
const secondaryColorDark = makeDarkerColor(secondaryColor);
const secondaryColorLight = makeLighterColor(secondaryColor);

export const defaultColorPalette: IDefaultColorPalette = {
  contrastDefaultLight,
  contrastDefaultDark,
  contrastText,
  primaryColor,
  primaryColorDark,
  primaryColorLight,
  secondaryColor,
  secondaryColorDark,
  secondaryColorLight,
};

export function makeDarkerColor(color: string): string {
  return chroma(color).darken().hex();
}

export function makeLighterColor(color: string): string {
  return chroma(color).alpha(0.1).hex();
}
