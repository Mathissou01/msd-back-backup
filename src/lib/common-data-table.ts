export interface ICommonDataTableValidation {
  isValid: boolean;
  errorMessage?: string;
}

export interface IDefaultTableRow {
  id: string;
  editState: boolean;
  expandableRow?: boolean;
}

export interface ILazyLoadingOptions {
  isRemote: boolean;
  totalRows: number;
}

export interface IPaginationOptions {
  hasPagination: boolean;
  hasRowsPerPageOptions?: boolean;
  defaultRowsPerPage: number;
  rowsPerPageOptions?: Array<number>;
  defaultPage?: number;
}

export interface ISortingParams {
  column: string;
  direction?: "asc" | "desc";
}

export interface ICurrentPagination {
  page: number;
  rowsPerPage: number;
  sort?: ISortingParams;
  filters?: Record<string, unknown>;
}
