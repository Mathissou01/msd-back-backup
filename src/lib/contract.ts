export enum EContractClientTypeLabels {
  city = "Commune",
  epci = "EPCI",
  union = "Syndicat",
}
