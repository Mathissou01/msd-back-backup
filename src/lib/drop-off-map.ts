import {
  CollectDropOff,
  CollectVoluntary,
} from "../graphql/codegen/generated-types";

export interface IDropOffMapStaticFields {
  name: string;
  phoneNumber?: string | null;
  mustKnow?: string | null;
  collectVoluntary: CollectVoluntary;
  collectDropOff: CollectDropOff;
  longitude: number;
  latitude: number;
}
