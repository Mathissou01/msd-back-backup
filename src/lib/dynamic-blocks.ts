import { v4 as uuidv4 } from "uuid";
import { Enum_Componentblockscommentary_Commentarystatus } from "../graphql/codegen/generated-types";
import { TBlockPictoStyles } from "./pictos";
import { IUploadFileEntity } from "./media";
import { removeNulls } from "./utilities";

export type TBlocksDynamicZone =
  | "ComponentBlocksFile"
  | "ComponentBlocksHorizontalRule"
  | "ComponentBlocksImage"
  | "ComponentBlocksSubHeading"
  | "ComponentBlocksVideo"
  | "ComponentBlocksWysiwyg"
  | "Error"
  // DropOffMap Blocks
  | "ComponentBlocksDownloadBlock"
  // Request Blocks
  | "ComponentBlocksAttachments"
  | "ComponentBlocksCommentary"
  | "ComponentBlocksCumbersome"
  | "ComponentBlocksQuestions"
  | "ComponentBlocksQcm"
  | "ComponentBlocksDateChoice"
  | "ComponentBlocksCheckbox"
  | "ComponentBlocksProofOfReceipt"
  | "ComponentBlocksRequestType";

export type TDynamicFieldOption = Exclude<TBlocksDynamicZone, "Error">;

export interface IDynamicFieldProps {
  minBlocks?: number;
  maxBlocks?: number;
}

export type TDynamicFieldConfiguration = {
  option: TDynamicFieldOption;
  props?: IDynamicFieldProps;
};

interface IBlockDisplayMap extends IDynamicFieldProps {
  label: string;
  picto: TBlockPictoStyles;
  isEmpty?: boolean;
}

export const blockDisplayMap: Record<TDynamicFieldOption, IBlockDisplayMap> = {
  ComponentBlocksFile: {
    label: "Fichier",
    picto: "attachment",
  },
  ComponentBlocksHorizontalRule: {
    label: "Séparateur",
    picto: "expandVertical",
    isEmpty: true,
  },
  ComponentBlocksImage: {
    label: "Image",
    picto: "picture",
  },
  ComponentBlocksSubHeading: {
    label: "Titre",
    picto: "text",
  },
  ComponentBlocksVideo: {
    label: "Vidéo",
    picto: "video",
  },
  ComponentBlocksWysiwyg: {
    label: "Texte",
    picto: "textCase",
  },
  // DropOffMap Blocks
  ComponentBlocksDownloadBlock: {
    label: "Fichier téléchargeable",
    picto: "attachment",
  },
  // Request Blocks
  ComponentBlocksAttachments: {
    label: "Pièces jointes",
    picto: "attachment",
  },
  ComponentBlocksCommentary: {
    label: "Commentaire",
    picto: "chatBubble",
  },
  ComponentBlocksCumbersome: {
    label: "TO_REPLACE_Cumbersome",
    picto: "paragraphJustified",
  },
  ComponentBlocksQuestions: {
    label: "Question texte",
    picto: "paragraphJustified",
  },
  ComponentBlocksQcm: {
    label: "Question à choix multiple",
    picto: "arrowRectangle",
  },
  ComponentBlocksDateChoice: {
    label: "Choix d'une date",
    picto: "calendar",
  },
  ComponentBlocksCheckbox: {
    label: "Case à cocher",
    picto: "checkboxIcon",
  },
  ComponentBlocksProofOfReceipt: {
    label: "TO_REPLACE_ProofOfReceipt",
    picto: "paragraphJustified",
  },
  ComponentBlocksRequestType: {
    label: "Type de demande",
    picto: "text",
  },
};

interface IPartialBlockDynamicZone {
  __typename: TBlocksDynamicZone;
  id: string;
}

interface IPartialBlock {
  __typename: TDynamicFieldOption;
  id: string;
}

export type IFormBlock =
  | IPartialBlock
  | IBlocksFile
  | IBlocksHorizontalRule
  | IBlocksImage
  | IBlocksSubHeading
  | IBlocksVideo
  | IBlocksWysiwyg
  | IBlocksAttachments
  | IBlocksDateChoice
  | IBlocksQCM
  | IBlocksQuestions
  | IBlocksCommentary
  | IBlocksCheckbox
  | IBlocksDownloadableFiles
  | IBlocksRequestType;

export interface IBlocksFile extends IPartialBlock {
  __typename: "ComponentBlocksFile";
  id: string;
  document: IUploadFileEntity;
}

export interface IBlocksHorizontalRule extends IPartialBlock {
  __typename: "ComponentBlocksHorizontalRule";
  hr: string;
}

export interface IBlocksImage extends IPartialBlock {
  __typename: "ComponentBlocksImage";
  altText: string;
  isDecorative: boolean;
  picture: IUploadFileEntity;
}

export interface IBlocksSubHeading extends IPartialBlock {
  __typename: "ComponentBlocksSubHeading";
  subHeadingTag?: string;
  subHeadingText?: string;
}

export interface IBlocksVideo extends IPartialBlock {
  __typename: "ComponentBlocksVideo";
  videoLink?: string;
  transcriptText?: string;
}

export interface IBlocksWysiwyg extends IPartialBlock {
  __typename: "ComponentBlocksWysiwyg";
  textEditor: string;
}

export interface IBlocksDownloadableFiles extends IPartialBlock {
  __typename: "ComponentBlocksDownloadBlock";
  linkText: string;
  file: IUploadFileEntity;
}

export interface IBlocksAttachments extends IPartialBlock {
  __typename: "ComponentBlocksAttachments";
  attachmentLabel: string;
  renderField: boolean;
  multipleAttachments: boolean;
}

export interface IBlocksQCM extends IPartialBlock {
  __typename: "ComponentBlocksQcm";
  fieldStatusQCM: string;
  fieldLabelQCM: string;
  responses: string;
  multipleChoice: string;
}

export interface IBlocksQuestions extends IPartialBlock {
  __typename: "ComponentBlocksQuestions";
  textStatus: string;
  questionTextLabel: string;
  questionTextPlaceholder: string;
  height: string;
}

export interface IBlocksCommentary extends IPartialBlock {
  __typename: "ComponentBlocksCommentary";
  commentaryStatus: Enum_Componentblockscommentary_Commentarystatus;
  commentaryLabel: string;
  commentaryPlaceholder: string;
}

export interface IBlocksCheckbox extends IPartialBlock {
  __typename: "ComponentBlocksCheckbox";
  fieldStatusCheckbox: string;
  labelCheckbox: string;
}

export interface IBlocksDateChoice extends IPartialBlock {
  __typename: "ComponentBlocksDateChoice";
  fieldStatus: string;
  fieldLabelDateChoice: string;
}

export interface IBlocksRequestType extends IPartialBlock {
  __typename: "ComponentBlocksRequestType";
  title?: string;
  isEmail: boolean;
  isTSMS: boolean;
  email: string;
}

/* Methods */
export function isFormBlock(
  block: Partial<IPartialBlockDynamicZone>,
): block is IFormBlock {
  return "__typename" in block && "id" in block;
}

export function remapFormBlocksDynamicZone(
  blocks?: Array<Partial<IPartialBlockDynamicZone> | null> | null,
): Array<IFormBlock> {
  return (
    blocks
      ?.map((block) => {
        if (block) {
          const type = block.__typename;
          if (type && type !== "Error" && isFormBlock(block)) {
            return {
              ...block,
              __typename: block.__typename,
              id: block.id,
            };
          }
        }
      })
      .filter(removeNulls) ?? []
  );
}

export function createEmptyBlock(__typename: TDynamicFieldOption): IFormBlock {
  const temporaryId = uuidv4();
  switch (__typename) {
    case "ComponentBlocksHorizontalRule": {
      return {
        ...{
          __typename,
          id: temporaryId,
        },
        hr: "<hr>",
      };
    }
    case "ComponentBlocksSubHeading": {
      return {
        __typename,
        id: temporaryId,
        subHeadingText: undefined,
        subHeadingTag: undefined,
      };
    }
    case "ComponentBlocksVideo": {
      return {
        __typename,
        id: temporaryId,
        videoLink: undefined,
        transcriptText: undefined,
      };
    }
    case "ComponentBlocksFile": {
      return {
        __typename,
        id: temporaryId,
        document: undefined,
      };
    }
    case "ComponentBlocksImage": {
      return {
        __typename,
        id: temporaryId,
        isDecorative: undefined,
        altText: undefined,
        picture: undefined,
      };
    }
    case "ComponentBlocksWysiwyg": {
      return {
        __typename,
        id: temporaryId,
        textEditor: undefined,
      };
    }
    case "ComponentBlocksQcm": {
      return {
        __typename,
        id: temporaryId,
        fieldStatusQCM: "Obligatoire",
        fieldLabelQCM: "",
        responses: "",
        multipleChoice: "0",
      };
    }
    case "ComponentBlocksQuestions": {
      return {
        __typename,
        id: temporaryId,
        textStatus: "Obligatoire",
        questionTextLabel: "",
        questionTextPlaceholder: "",
        height: "0",
      };
    }
    case "ComponentBlocksAttachments": {
      return {
        __typename,
        id: temporaryId,
        attachmentLabel: "",
        renderField: false,
        multipleAttachments: false,
      };
    }
    case "ComponentBlocksCommentary": {
      return {
        __typename,
        id: temporaryId,
        commentaryStatus:
          Enum_Componentblockscommentary_Commentarystatus.Facultatif,
        commentaryLabel: undefined,
        commentaryPlaceholder: undefined,
      };
    }
    case "ComponentBlocksCheckbox": {
      return {
        __typename,
        id: temporaryId,
        fieldStatusCheckbox: "Obligatoire",
        labelCheckbox: "",
      };
    }
    case "ComponentBlocksDownloadBlock": {
      return {
        __typename,
        id: temporaryId,
        linkText: undefined,
        file: undefined,
      };
    }
    case "ComponentBlocksRequestType": {
      return {
        __typename,
        id: temporaryId,
        title: "",
        isEmail: false,
        isTSMS: false,
        email: "",
      };
    }
    case "ComponentBlocksDateChoice": {
      return {
        __typename,
        id: temporaryId,
        fieldStatus: "Obligatoire",
        fieldLabelDateChoice: "",
      };
    }
    default: {
      return { __typename, id: temporaryId };
    }
  }
}

/**
 * Function to use minBlock behavior (to generate missing blocks matching your blockConfigurations from your fields data)
 * @param blockConfigurations Array<TDynamicFieldConfiguration>
 * @param fields Array<IPartialBlock>
 * @returns Array<IFormBlock>
 */
export function generateMinimumBlocks(
  blockConfigurations: Array<TDynamicFieldConfiguration>,
  fields: Array<IPartialBlock>,
): Array<IFormBlock> {
  blockConfigurations.forEach((blockConfiguration) => {
    if (blockConfiguration.props?.minBlocks) {
      const minBlocks = blockConfiguration.props?.minBlocks;
      const fieldsOfTypename = fields.filter((field) => {
        return field.__typename === blockConfiguration.option;
      });
      if (fieldsOfTypename.length < minBlocks) {
        for (let i = fieldsOfTypename.length; i < minBlocks; i++) {
          fields.push(createEmptyBlock(blockConfiguration.option));
        }
      }
    }
  });
  return fields;
}
