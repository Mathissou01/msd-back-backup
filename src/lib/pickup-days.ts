enum EPeriodicityStatus {
  WEEKLY = "hebdomadaire",
  MONTHLY = "mensuel",
}

enum EMonthlyStatus {
  MONTHLY_FIRST = "le premier",
  MONTHLY_LAST = "le dernier",
  MONTHLY_DATE = "le N du mois",
}

enum EDaysOfTheWeek {
  MONDAY = "Lundi",
  TUESDAY = "Mardi",
  WEDNESDAY = "Mercredi",
  THURSDAY = "Jeudi",
  FRIDAY = "Vendredi",
  SATURDAY = "Samedi",
  SUNDAY = "Dimanche",
}

interface IHebdomadireAdvancedSelection {
  hebdomadaire: {
    selection: Array<string>;
  };
}

interface IMensuelAdvancedSelection {
  mensuel: {
    choice: string;
    selection: Array<string> | number;
  };
}

const dayOptions = [
  { label: EDaysOfTheWeek.MONDAY, value: EDaysOfTheWeek.MONDAY },
  { label: EDaysOfTheWeek.TUESDAY, value: EDaysOfTheWeek.TUESDAY },
  { label: EDaysOfTheWeek.WEDNESDAY, value: EDaysOfTheWeek.WEDNESDAY },
  { label: EDaysOfTheWeek.THURSDAY, value: EDaysOfTheWeek.THURSDAY },
  { label: EDaysOfTheWeek.FRIDAY, value: EDaysOfTheWeek.FRIDAY },
  { label: EDaysOfTheWeek.SATURDAY, value: EDaysOfTheWeek.SATURDAY },
  { label: EDaysOfTheWeek.SUNDAY, value: EDaysOfTheWeek.SUNDAY },
];

const recurrenceOptions = [
  {
    option: EMonthlyStatus.MONTHLY_FIRST,
    label: EMonthlyStatus.MONTHLY_FIRST,
  },
  {
    option: EMonthlyStatus.MONTHLY_LAST,
    label: EMonthlyStatus.MONTHLY_LAST,
  },
  {
    option: EMonthlyStatus.MONTHLY_DATE,
    label: EMonthlyStatus.MONTHLY_DATE,
  },
];

const periodicityOptions = [
  {
    option: EPeriodicityStatus.WEEKLY,
    label: EPeriodicityStatus.WEEKLY,
  },
  {
    option: EPeriodicityStatus.MONTHLY,
    label: EPeriodicityStatus.MONTHLY,
  },
];

export {
  EPeriodicityStatus,
  EMonthlyStatus,
  EDaysOfTheWeek,
  dayOptions,
  recurrenceOptions,
  periodicityOptions,
};
export type { IHebdomadireAdvancedSelection, IMensuelAdvancedSelection };
