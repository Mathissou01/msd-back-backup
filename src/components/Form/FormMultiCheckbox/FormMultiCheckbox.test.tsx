import React, { ReactNode } from "react";
import { FormProvider, useForm } from "react-hook-form";
import { fireEvent, render, screen } from "@testing-library/react";
import FormMultiCheckbox from "./FormMultiCheckbox";
import { dayOptions } from "../../../lib/pickup-days";

const mock = {
  name: "form-checkbox",
  label: "form checkbox",
};

const Wrapper = (props: { children: ReactNode }) => {
  const formMethods = useForm();
  return <FormProvider {...formMethods}>{props.children}</FormProvider>;
};

describe("FormMultiCheckbox", () => {
  it("renders", () => {
    const { container } = render(
      <Wrapper>
        <FormMultiCheckbox
          name={mock.name}
          label={mock.label}
          options={dayOptions}
          isRequired
        />
      </Wrapper>,
    );

    const checkbox = screen.getByTestId("form-multi-checkbox_0");
    expect(checkbox).toBeInTheDocument();
    expect(checkbox).not.toHaveClass("c-FormMultiCheckbox__Input_checked");

    expect(container).toMatchSnapshot();
  });

  it("changes style when checked", async () => {
    render(
      <Wrapper>
        <FormMultiCheckbox
          name={mock.name}
          label={mock.label}
          options={dayOptions}
          isRequired
        />
      </Wrapper>,
    );

    const checkbox = screen.getByTestId("form-multi-checkbox_0");
    expect(checkbox).not.toHaveClass("c-FormMultiCheckbox__Input_checked");
    fireEvent.click(checkbox);
    expect(checkbox).toHaveClass("c-FormMultiCheckbox__Input_checked");
  });

  it("can be disabled", () => {
    render(
      <Wrapper>
        <FormMultiCheckbox
          name={mock.name}
          label={mock.label}
          options={dayOptions}
          isRequired
          isDisabled={true}
        />
      </Wrapper>,
    );

    const checkbox = screen.getByTestId("form-multi-checkbox_0");
    expect(checkbox).toHaveAttribute("disabled");
    fireEvent.click(checkbox);
    expect(checkbox).not.toHaveClass("c-FormMultiCheckbox__Input_checked");
  });
});
