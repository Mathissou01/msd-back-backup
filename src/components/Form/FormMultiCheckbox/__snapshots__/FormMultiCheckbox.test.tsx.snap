// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormMultiCheckbox renders 1`] = `
<div>
  <div
    class="c-FormMultiCheckbox"
  >
    <fieldset>
      <div
        class="c-FormLabel"
      >
        <legend
          class="c-FormLabel__LabelWrapper"
        >
          <span
            class="c-FormLabel__Label"
          >
            form checkbox *
          </span>
        </legend>
      </div>
      <div
        class="c-FormMultiCheckbox__Options c-FormMultiCheckbox__Options_horizontal"
      >
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_0"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Lundi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Lundi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_1"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Mardi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Mardi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_2"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Mercredi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Mercredi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_3"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Jeudi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Jeudi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_4"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Vendredi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Vendredi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_5"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Samedi"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Samedi
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-FormMultiCheckbox__Option"
        >
          <input
            class="c-FormMultiCheckbox__Input "
            data-testid="form-multi-checkbox_6"
            id="form-checkbox"
            name="form-checkbox"
            type="checkbox"
            value="Dimanche"
          />
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__LabelWrapper"
              for="form-checkbox"
            >
              <span
                class="c-FormLabel__Label"
              >
                Dimanche
              </span>
            </label>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  ,
</div>
`;
